src/PostgrestFilterBuilder.ts

```diff
@@ -73,9 +73,11 @@ export default class PostgrestFilterBuilder<
    */
   eq<ColumnName extends string>(
     column: ColumnName,
-    value: ResolveFilterValue<Schema, Row, ColumnName> extends never
-      ? NonNullable<unknown>
-      : NonNullable<ResolveFilterValue<Schema, Row, ColumnName>>
+    value: ResolveFilterValue<Schema, Row, ColumnName> extends infer ResolvedFilterValue
+      ? ResolvedFilterValue extends never
+        ? NonNullable<unknown>
+        : NonNullable<ResolvedFilterValue>
+      : never
```

Comment:
The actual tricks is to use the `infer` and `never` to force typescript to infer the right type and not stop at "possibly infinite recursive".